noir-projects/noir-contracts/contracts/inclusion_proofs_contract/src/test.nr

```diff
@@ -0,0 +1,366 @@
+use crate::InclusionProofs;
+use dep::aztec::{
+    oracle::execution::get_contract_address,
+    prelude::AztecAddress,
+    test::helpers::{cheatcodes, test_environment::TestEnvironment},
+};
+
+global INITIAL_VALUE: Field = 69;
+
+pub unconstrained fn setup(
+    initial_value: Field,
+) -> (&mut TestEnvironment, AztecAddress, AztecAddress) {
+    // Setup env, generate keys
+    let mut env = TestEnvironment::new();
+    let owner = env.create_account(1);
+    env.impersonate(owner);
+
+    // We advance one block here, because we want to deploy our contract in block 3.
+    // This is because we will do tests later that prove the non inclusion of values and of the contract itself at block 2.
+    env.advance_block_by(1);
+
+    // Deploy contract and initialize
+    let initializer = InclusionProofs::interface().constructor(initial_value);
+    let inclusion_proofs_contract =
+        env.deploy_self("InclusionProofs").with_public_void_initializer(initializer);
+    let contract_address = inclusion_proofs_contract.to_address();
+    env.advance_block_by(1);
+    (&mut env, contract_address, owner)
+}
+
+#[test]
+unconstrained fn note_flow() {
+    // This test creates a note and checks certain properties about it, namely its inclusion in the trees,
+    // the non-inclusion of its nullifier, and its validity (the combination of the previous two)
+    let (env, contract_address, owner) = setup(INITIAL_VALUE);
+    env.impersonate(owner);
+
+    let NOTE_VALUE = 69;
+    InclusionProofs::at(contract_address).create_note(owner, NOTE_VALUE).call(&mut env.private());
+
+    env.advance_block_by(1);
+
+    let note_creation_block_number = env.block_number();
+
+    // We advance by another block to make sure that the note creation block number != our current block number
+    env.advance_block_by(1);
+
+    let current_contract_address = get_contract_address();
+    cheatcodes::set_contract_address(contract_address);
+
+    // We fetch the note we created and make sure that it is valid.
+    let note = InclusionProofs::get_note(owner);
+    cheatcodes::set_contract_address(current_contract_address);
+
+    assert(note.owner.eq(owner));
+    assert(note.value.eq(NOTE_VALUE));
+
+    // Each of these tests (note inclusion, note non-nullification, and validity (inclusion & non-nullification)) check the assertion at the block of creation of note, as well as at the "current" block
```

Comment:
I see that block with number 0 is used. Is that the current block? That's kinda weird. Also as far as I remember block 0 is not a legitimate block number (we start from 1).

Reply:
i see that the InclusionProof contract ignores it.

Please nuke the `use_block_number` param from InclusionProofs contract and always use the block number.

It used to be there because from TS we don't have a direct control over block numbers (which we do in TXE tests)